app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        func_name = func.__name__
+        if func_name not in results:
+            results[func_name] = {}
+        if args in results[func_name]:
+            print("Getting from cache")
+            return results[func_name][args]
+        else:
+            print("Calculating new result")
+            results[func_name][args] = func(*args)
+            return results[func_name][args]
+    return wrapper
```

Comment:
Your results dictionary can use the args tuple as a key, rather than nesting it within another dictionary keyed by function name.